### PR TITLE
add ruby tests 3.0 and 3.1 in verify pipeline

### PR DESCRIPTION
**Pull request by jayashrig158**
Signed-off-by: jayashri garud <jgarud@msystechnologies.com>

### CODE REVIEW

**Comment by jayashrig158** [comment]
@tas50 , Please suggest if we can remove ruby 2.4 support here as well.

**Comment by tas50** [comment]
We can remove Ruby 2.4/2.5 here

File: Gemfile [hunk 1]
```diff
@@ -1,23 +1,25 @@
-source "https://rubygems.org"
+# frozen_string_literal: true
```
Comment (PrajaktaPurohit) [inline Gemfile:1]: Have these been through rubocop instead of chefstyle?

Comment (jayashrig158) [reply Gemfile:1]: @PrajaktaPurohit  Yes,these have been through rubocop.

Comment (PrajaktaPurohit) [reply Gemfile:1]: We need to run these through chefstyle instead to get the linter to pass.

Comment (lamont-granquist) [reply Gemfile:1]: Yeah plain rubocop adds the frozen_string_literal comments which broke everything and why we don't have that turned on in chefstyle.

All the chef repos that aren't cookbooks need to be run through chefstyle, not rubocop (or cookstyle, which is for cookbooks-only)

**Review by PrajaktaPurohit — APPROVED** [review]

**Review by skeshari12 — APPROVED** [review]

File: spec/mixlib/cli_spec.rb [hunk 2]
```diff
@@ -465,6 +465,7 @@ class T2 < TestCLI
       class T3 < TestCLI
         option :arg2, boolean: true
       end
+
```
[the remark this answers is not in the document]
Comment (jayashrig158) [reply spec/mixlib/cli_spec.rb:468]: @PrajaktaPurohit  This was rubocop offences correction.

**Pull request merged**